### PR TITLE
feat: add @talend/html-webpack-plugin

### PR DESCRIPTION
**Pull request by jmfrancois**
**What is the problem this PR is trying to solve?**

AppLoader need lots of setup in webpack to tune the html template

BUG: the app loader rely on rem unit for CSS. so until the CSS of the APP is loaded we rely on default font size. So when the APP.css is loaded the loader size change ...

**What is the chosen solution to this problem?**

Let's add a webpack plugin to hide the complexity

Move the AppLoader style size to fixed pixels

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/html-webpack-plugin/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# @talend/html-webpack-plugin
+
+This plugin is to provide some default [hooks](https://github.com/jantimon/html-webpack-plugin#events) into `html-webpack-plugin`
+
+## options
+
+The plugin support the following options:
+
+| name | description |
```
Comment (annilsson) [inline packages/html-webpack-plugin/README.md]: Don't forget to update this. 2 options missing `appLoaderIcon` and `versions`. And I think one was removed, `headerStyles`?

File: packages/html-webpack-plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+```javascript
+const TalendHTML = require('@talend/html-webpack-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const Loader = require('@talend/react-components/lib/AppLoader/constant').default;
+const ICON = "url('data:image/svg+xml;base64,PHN2...')";
+const EXTENSION = process.env.NODE_ENV === ENV.DEV ? 'js' : 'min.js';
+
+//... plugins
+    new HtmlWebpackPlugin({
+        // ... your options
```
Comment (annilsson) [inline packages/html-webpack-plugin/README.md]: Maybe we can also add the loader key here, along with a very short index.html to show how you can insert the loader markup? Just to make sure we all use it the same way

File: packages/html-webpack-plugin/README.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    }),
+    new TalendHTML({
```
Comment (annilsson) [inline packages/html-webpack-plugin/README.md]: Don't forget to update example if you update the options

File: packages/html-webpack-plugin/index.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+				data.head = data.head.map(head => {
+					if (head.tagName !== 'link') {
+						return head;
+					}
+					const media = head.attributes.media || 'all';
+					// eslint-disable-next-line no-param-reassign
+					head.attributes.media = 'none';
+					// eslint-disable-next-line no-param-reassign
+					head.attributes.onload = `media='${media}'`;
+					// data.body.splice(index, 0, head);
```
Comment (romainseb) [inline packages/html-webpack-plugin/index.js]: Could we remove that comment as it seems to not be needed anymore ?

File: packages/html-webpack-plugin/index.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+				});
+			}
+			if (options.loadCSSAsync) {
+				// eslint-disable-next-line no-param-reassign
+				data.head = data.head.map(head => {
+					if (head.tagName !== 'link') {
+						return head;
+					}
+					const media = head.attributes.media || 'all';
+					// eslint-disable-next-line no-param-reassign
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/index.js]: It seems fair to disable the rule for the webpack plugin, what do you think ?

File: packages/html-webpack-plugin/package.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  "repository": {
+    "type": "git",
+    "url": "github:Talend/ui"
+  },
+  "devDependencies": {
+    "eslint": "3.4.0",
+    "eslint-config-airbnb": "10.0.1",
+    "eslint-plugin-import": "1.14.0",
+    "jest-cli": "20.0.3",
+    "webpack": "1.13.1"
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/package.json]: `webpack` and `jest` are not useful as dependencies here.

File: packages/html-webpack-plugin/package.json [hunk 7]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "@talend/html-webpack-plugin",
+  "version": "0.0.1",
+  "description": "@talend/html-webpack-plugin library.",
+  "license": "Apache-2.0",
+  "author": "jmfrancois jmfrancois@talend.com",
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/package.json]: This should be talend frontend mail, look at the other packages

File: packages/html-webpack-plugin/package.json (hunk 6 above)
Comment (jmfrancois) [reply packages/html-webpack-plugin/package.json]: I have tests based on jest

File: packages/html-webpack-plugin/index.test.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+			closeTag: true,
+			attributes: { rel: 'stylesheet', type: 'text/css', href: 'app.css' },
+		},
+	],
+	body: [
+		{ tagName: 'script', closeTag: true, attributes: { type: 'text/javascript', href: 'app.js' } },
+	],
+};
+
+const OPTIONS = {
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/index.test.js]: This is never used

File: packages/html-webpack-plugin/index.test.js [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+			media: 'none',
+			onload: 'media=\'all\'',
+		});
+	});
+	it('should call with option versions to add TALEND_APP_INFO global', () => {
+		refresh({ versions: { '@talend/my-app': '1.2.3' } });
+		fnc(DATA, callback);
+		expect(callback.mock.calls[0][1].body[0]).toMatchObject({
+			tagName: 'script',
+			innerHTML: 'TALEND_APP_INFO={\"@talend/my-app\":\"1.2.3\"}',
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/index.test.js]: You don't need to escape double quotes

File: packages/html-webpack-plugin/index.test.js [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+		};
+		plugin.apply(compiler);
+		compilation = compiler.plugin.mock.calls[0][1];
+		compilation(compiler);
+		fnc = compiler.plugin.mock.calls[1][1];
+		callback = jest.fn();
+	}
+
+	beforeEach(() => {
+		refresh();
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/index.test.js]: This is a little overkill, it is useful for 2 tests. Please just call it in your 2 tests

File: packages/html-webpack-plugin/index.test.js [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+const OPTIONS = {
+	bodyBefore: [],
+};
+
+describe('@talend/html-webpack-plugin', () => {
+	let plugin;
+	let compiler;
+	let compilation;
+	let fnc;
+	let callback;
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/index.test.js]: Can you rename those 2 vars to something more explicit ?
`fnc` --> `pluginExecFn` ?
`callback` --> `pluginExecCallback` ?

File: packages/html-webpack-plugin/index.test.js [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+		refresh({ foo: 'bar' });
+		expect(plugin.options.foo).toBe('bar');
+	});
+	it('should apply call compiler/plugin function', () => {
+		expect(compiler.plugin).toHaveBeenCalled();
+		expect(compiler.plugin.mock.calls[0][0]).toBe('compilation');
+		expect(compiler.plugin.mock.calls[1][0]).toBe('html-webpack-plugin-alter-asset-tags');
+		expect(typeof fnc).toBe('function');
+	});
+	it('should call with DATA without options do not modify them', () => {
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/index.test.js]: This sentence is not correct --> `should not modify data when no option is provided` ?

File: packages/html-webpack-plugin/index.test.js [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+		fnc(DATA, callback);
+		expect(callback).toHaveBeenCalledWith(null, DATA);
+	});
+	it('should call with option bodyBefore add it to the content', () => {
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/index.test.js]: Same here --> `should add data before content based on bodyBefore option` ?

File: packages/html-webpack-plugin/index.test.js [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+		const item = { foo: 'bar' };
+		refresh({ bodyBefore: [item] });
+		fnc(DATA, callback);
+		expect(callback.mock.calls[0][1].body[0]).toBe(item);
+	});
+	it('should call with option loadCSSAsync modify media', () => {
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/index.test.js]: `should modify <link> media with loadCSSAsync option`

File: packages/html-webpack-plugin/index.test.js [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+		refresh({ loadCSSAsync: true });
+		fnc(DATA, callback);
+		expect(callback.mock.calls[0][1].head[0].attributes).toMatchObject({
+			media: 'none',
+			onload: 'media=\'all\'',
+		});
+	});
+	it('should call with option versions to add TALEND_APP_INFO global', () => {
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/index.test.js]: `should add TALEND_APP_INFO global var with versions option`

File: packages/html-webpack-plugin/index.test.js [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+		refresh({ versions: { '@talend/my-app': '1.2.3' } });
+		fnc(DATA, callback);
+		expect(callback.mock.calls[0][1].body[0]).toMatchObject({
+			tagName: 'script',
+			innerHTML: 'TALEND_APP_INFO={\"@talend/my-app\":\"1.2.3\"}',
+		});
+	});
+	it('should call with option appLoaderIcon to add inline style in head', () => {
```
Comment (jsomsanith) [inline packages/html-webpack-plugin/index.test.js]: `should inline style in head with appLoaderIcon option`

**Pull request merged**